disputer/index.js

```diff
@@ -24,73 +24,76 @@ const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
+
+    // Setup price feed.
+    // TODO: consider making getTime async and using contract time.
+    const getTime = () => Math.round(new Date().getTime() / 1000);
+    const priceFeed = await createPriceFeed(Logger, web3, new Networker(Logger), getTime, priceFeedConfig);
+
+    if (!priceFeed) {
+      throw "Price feed config is invalid";
+    }
 
-  // Client and dispute bot
-  const empClient = new ExpiringMultiPartyClient(Logger, ExpiringMultiParty.abi, web3, emp.address);
-  const gasEstimator = new GasEstimator(Logger);
-  const disputer = new Disputer(Logger, empClient, gasEstimator, priceFeed, accounts[0]);
+    // Client and dispute bot
+    const empClient = new ExpiringMultiPartyClient(Logger, ExpiringMultiParty.abi, web3, emp.address);
+    const gasEstimator = new GasEstimator(Logger);
+    const disputer = new Disputer(Logger, empClient, gasEstimator, priceFeed, accounts[0]);
 
-  while (true) {
-    try {
+    while (true) {
       await disputer.queryAndDispute();
       await disputer.queryAndWithdrawRewards();
-    } catch (error) {
-      Logger.error({
-        at: "Disputer#index🚨",
-        message: "Disputer error",
-        error: error
-      });
-    }
-    await delay(Number(pollingDelay));
 
-    if (!shouldPoll) {
-      break;
+      await delay(Number(pollingDelay));
+
+      if (!shouldPoll) {
+        break;
+      }
     }
+  } catch (error) {
+    Logger.error({
+      at: "Disputer#index🚨",
+      message: "Disputer error",
+      error: error.toString()
+    });
+    await delay(5000); // Hacky fix to ensure that winston still fires messages upstream.
   }
 }
 
 const Poll = async function(callback) {
   try {
     if (!process.env.EMP_ADDRESS) {
-      throw new Error(
-        "Bad input arg! Specify an `EMP_ADDRESS ` for the location of the expiring Multi Party within your environment variables."
-      );
+      throw "Bad input arg! Specify an `EMP_ADDRESS` for the location of the expiring Multi Party within your environment variables.";
     }
 
     const pollingDelay = process.env.POLLING_DELAY ? process.env.POLLING_DELAY : 10000;
 
     if (!process.env.PRICE_FEED_CONFIG) {
-      throw new Error(
-        "Bad input arg! Specify an `PRICE_FEED_CONFIG ` for the location of the expiring Multi Party within your environment variables."
-      );
+      throw "Bad input arg! Specify an `PRICE_FEED_CONFIG` for the location of the expiring Multi Party within your environment variables.";
     }
 
     const priceFeedConfig = JSON.parse(process.env.PRICE_FEED_CONFIG);
 
     await run(process.env.EMP_ADDRESS, true, pollingDelay, priceFeedConfig);
-  } catch (err) {
-    console.error(err);
-    callback(err);
+  } catch (error) {
+    Logger.error({
+      at: "Disputer#index🚨",
+      message: "Disputer configuration error",
+      error: error.toString()
+    });
+    await delay(5000); // Hacky fix to ensure that winston still fires messages upstream.
+    callback(error);
```

Comment:
> > > > Fixes 1,2,3,5 look great. I am a bit uncomfortable with that 5-second delay fix, which you note is hacky. Is there a way that we can pass a callback to the Logger so that downstream modules can know when the PagerDuty/Slack events are fired?
> > > 
> > > 
> > > ye, this is possible. I thought about this a bunch and also tried something like this. Ideally, you don't want the thread to yield until the async returns. however, this is actually quite complex to implement (unless there is a cleaner solution I'm not seeing) and so I thought it best to implement this "hacky" solution for now and we can refine it later when we have more time.
> > 
> > 
> > Yeah makes sense I'm sure you tried to think of alternative solutions. @mrice32 any ideas?
> 
> As mentioned IRL, can we do this? [winstonjs/winston#awaiting-logs-to-be-written-in-winston](https://github.com/winstonjs/winston#awaiting-logs-to-be-written-in-winston)
> 
> PR looks good outside of that!

I tried this but it doesn't actually solve our problem, unfortunately. Wrapping the `callback` in a `Logger.on("finish", function(info) {})` still does not yield a slack message.